### DISCUSSION TITLE
Source-controller using mTLS (mutual authentication)

### DISCUSSION THREAD

**Discussion by alexisbonnin** [fluxcd/flux2#704] [Proposals] [answered]
Hi there,

As Flux v1 uses git binary to perform cloning, it was possible to mount a volume on Flux operator to provide self signed certificates and set ./root/gitconfig to use those certificates in order to perform mutual authentication.

As source-controller now uses golang lib / alternative git library, it is not possible provide certificates the same way.

As anyone been able to perform mTLS on source controller ? 

Would it be possible to update source-controller to be able to perform mTLS ?

Regards

**Comment by stefanprodan** [comment, marked as answer]
> Would it be possible to update source-controller to be able to perform mTLS ?

Yes, here is the issues that tracks this feature https://github.com/fluxcd/source-controller/issues/93
